falcon/api.py

```diff
@@ -615,26 +578,37 @@ def _get_body(self, resp, wsgi_file_wrapper=None):
                 when resp.stream is a file-like object (default None).
 
         Returns:
-            * If resp.body is not ``None``, returns [resp.body], encoded
-              as UTF-8 if it is a Unicode string. Bytestrings are returned
-              as-is.
-            * If resp.data is not ``None``, returns [resp.data]
-            * If resp.stream is not ``None``, returns resp.stream
-              iterable using wsgi.file_wrapper, if possible.
-            * Otherwise, returns []
+            A two-member tuple of the form (iterable, content_length). The
+            length is returned as -1 when unknown. The iterable is determined
+            as follows:
+
+                * If resp.body is not ``None``, returns [resp.body], encoded
+                  as UTF-8 if it is a Unicode string. Bytestrings are returned
+                  as-is.
+                * If resp.data is not ``None``, returns [resp.data]
+                * If resp.stream is not ``None``, returns resp.stream
+                  iterable using wsgi.file_wrapper, if possible.
+                * Otherwise, returns []
 
         """
 
-        body = resp.body_encoded
-
+        body = resp._get_encoded_body()
```

Comment:
Perhaps I should just get rid of `_get_encoded_boy()` altogether and put the logic here instead?